aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java

```diff
@@ -687,6 +691,197 @@ public void onError(Exception error) {
+                    userAttributeMap,
+                    new Callback<List<UserCodeDeliveryDetails>>() {
+                        @Override
+                        public void onResult(List<UserCodeDeliveryDetails> result) {
+                            if (result != null) {
+                                for (UserCodeDeliveryDetails details : result) {
+                                    if (details.getAttributeName().equals(userAttributeKeyString)) {
+                                        Map<AuthUserAttributeKey, AuthUpdateAttributeResult> resultMap =
+                                                new HashMap<>();
+                                        resultMap.put(new AuthUserAttributeKey(userAttributeKeyString),
+                                                new AuthUpdateAttributeResult(
+                                                        true,
+                                                        new AuthNextUpdateAttributeStep(
+                                                                AuthUpdateAttributeStep.CONFIRM_ATTRIBUTE_WITH_CODE,
+                                                                Collections.emptyMap(),
+                                                                convertCodeDeliveryDetails(details))
+                                                ));
+
+                                        onSuccess.accept(resultMap);
+                                    }
+                                }
+                            } else {
+                                onError.accept(new AuthException(
+                                        "Code delivery failed",
+                                        "See attached exception for more details"
+                                ));
+                            }
+                        }
+
+                        @Override
+                        public void onError(Exception error) {
+                            onError.accept(new AuthException(
+                                            "Failed to update user attribute",
+                                            error,
+                                            "See attached exception for more details"
+                                    )
+                            );
+                        }
+                    }
+            );
+        }
+
+    }
+
+    @Override
+    public void resendUserAttributeConfirmationCode(
+            @NonNull AuthUserAttributeKey attributeKey,
+            @NonNull Consumer<AuthCodeDeliveryDetails> onSuccess,
+            @NonNull Consumer<AuthException> onError
+    ) {
+        String attributeName = attributeKey.getKeyString();
+        awsMobileClient.verifyUserAttribute(attributeName, new Callback<UserCodeDeliveryDetails>() {
+            @Override
+            public void onResult(UserCodeDeliveryDetails result) {
+                if (result.getAttributeName().equals(attributeName)) {
```

Comment:
Again, you need to handle the case where it doesn't equal it - perhaps returning an error if that's proper. But currently it'll just hang and never return a success or error callback if this condition is false.